File: Server/route/group.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetGroupRoute(r *gin.Engine) {
	adminGroup := r.Group("/group")
	adminGroup.Use(middleware.AdminAuthMiddleware())
	{
		adminGroup.POST("/GetGroupList", api.GetGroupList)
		adminGroup.POST("/GetGroup", api.GetGroup)
		adminGroup.POST("/CreateGroup", api.CreateGroup)
		adminGroup.POST("/UpdateGroup", api.UpdateGroup)
		adminGroup.POST("/DeleteGroup", api.DeleteGroup)
	}
}

File: Web/src/api/group.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetGroupList() {
    return apiClient.post("/group/GetGroupList").then(handleResponse).catch(handleError)
}

function GetGroupListByQuery(data) {
  return apiClient.post("/group/GetGroupListByQuery", data).then(handleResponse).catch(handleError)
}

function GetGroup(data) {
    return apiClient.post("/group/GetGroup", data).then(handleResponse).catch(handleError)
}

function CreateGroup(data) {
    return apiClient.post("/group/CreateGroup", data).then(handleResponse).catch(handleError)
}

function UpdateGroup(data) {
    return apiClient.post("/group/UpdateGroup", data).then(handleResponse).catch(handleError)
}

function DeleteGroup(data) {
    return apiClient.post("/group/DeleteGroup", data).then(handleResponse).catch(handleError)
}

const groupApi = {
  GetGroupList,
  GetGroupListByQuery,
  GetGroup,
  CreateGroup,
  UpdateGroup,
  DeleteGroup
}

export default groupApi
